central/auth/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,315 @@
+
+	s.NoError(s.roleDataStore.RemoveRole(s.ctx, testRole1))
+}
+
+func (s *datastorePostgresTestSuite) TestAddGitHubActionConfigFillIssuer() {
+	config, err := s.authDataStore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		Type:                    storage.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+		TokenExpirationDuration: "5m",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "some-value",
+				Role:  testRole1,
+			},
+		},
+	})
+
+	s.NoError(err)
+
+	s.Equal("https://token.actions.githubusercontent.com", config.GetIssuer())
+}
+
+func (s *datastorePostgresTestSuite) TestAddGitHubActionConfigCustomIssuer() {
+	config, err := s.authDataStore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		Type:                    storage.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+		TokenExpirationDuration: "5m",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "some-value",
+				Role:  testRole1,
+			},
+		},
+		Issuer: "https://stackrox.io",
+	})
+
+	s.NoError(err)
+
+	s.Equal("https://stackrox.io", config.GetIssuer())
+}
+
+func (s *datastorePostgresTestSuite) TestUpdateGitHubActionConfigFillIssuer() {
+	config, err := s.authDataStore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		Type:                    storage.AuthMachineToMachineConfig_GENERIC,
+		TokenExpirationDuration: "5m",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "some-value",
+				Role:  testRole1,
+			},
+		},
+		Issuer: "https://stackrox.io",
+	})
+
+	s.NoError(err)
+
+	config.Type = storage.AuthMachineToMachineConfig_GITHUB_ACTIONS
+	config.Issuer = ""
+
+	err = s.authDataStore.UpdateAuthM2MConfig(s.ctx, config)
+	s.NoError(err)
+
+	updatedConfig, exists, err := s.authDataStore.GetAuthM2MConfig(s.ctx, config.GetId())
+	s.NoError(err)
+	s.True(exists)
+	s.Equal("https://token.actions.githubusercontent.com", updatedConfig.GetIssuer())
+}
+
+func (s *datastorePostgresTestSuite) TestUpdateGitHubActionConfigCustomIssuer() {
+	config, err := s.authDataStore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		Type:                    storage.AuthMachineToMachineConfig_GENERIC,
+		TokenExpirationDuration: "5m",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "some-value",
+				Role:  testRole1,
+			},
+		},
+		Issuer: "https://stackrox.io",
+	})
+
+	s.NoError(err)
+
+	config.Type = storage.AuthMachineToMachineConfig_GITHUB_ACTIONS
+	config.Issuer = "https://private.stackrox.io"
+
+	err = s.authDataStore.UpdateAuthM2MConfig(s.ctx, config)
+	s.NoError(err)
```

Comment:
I keep running into this (https://github.com/stackrox/stackrox/pull/8145#discussion_r1366755178) and it keeps confusing me. For me, as a hypothetical user, the confusion is: does this `https://private.stackrox.io/` url have any effect? I could set it because the system did not reject `https://private.stackrox.io/`. Now what? Will it expect `https://private.stackrox.io/` or `https://token.actions.githubusercontent.com` in tokens?

Reply:
Reverted it, it's not possible to set this to something different anymore and `https://token.actions.githubusercontent.com` will be forced when `type=GITHUB_ACTIONS`.

Reply:
Thanks. I hope that you found my reasons reasonable and it's not just because of the volume of comments that I sent about and the time pressure.

Reply:
No it's reasonable, I think if we add support for GitHub enterprise / cloud it probably should be just a different type (OTOH I don't know what specialities that might involve, so better to be on the safe side and disallow it).